docs/vpm-spec.md

```diff
@@ -0,0 +1,428 @@
+
+## 1. Introduction
+### 1.1. Purpose
+The VRChat Package Manager is a package manager developed by VRChat and widely used by VRChat community.
+However, the VRChat Creator Companion (VCC), the official VPM implementation, has several bugs that make it hard to know the VPM Client should work.
+Therefore, vrc-get developers assume the best VPM behavior based on the behavior of VCC, and [[VCC Docs]].
+This document describes the vrc-get developers' understanding of the VPM and vrc-get extensions.
+
+### 1.2. Core Concepts
+The VPM provides a simple way to distribute UPM packages.
+
+The VPM Repository provides a list of packages and their metadata, and the link to the package archive file, in JSON document hosted on the HTTP server.
+The VPM Repository can authorize VPM Clients by using Header-based authentication, including downloading the package archive file.
+
+The VPM Client fetches the package information from Package Repositories and collects packages to install based on requested packages, version constraints, and dependencies.
+The VPM Client then downloads the package archives from the URL provided by the Repository and extracts the package archive into `Packages` folder in the Unity project.
+The VPM Client also writes the managing package information to `Packages/vpm-manifest.json` to manage the installed packages.
+The VPM Client can re-install the package by using the information.
+This information is usually used to install packages after cloning the project with Version Control System.
+VPM Clients can share the user-wide installed repositories list and local package cache for better user experience.
+
+## 2. Conformance
+### 2.1. Syntax Notation
+This specification uses the Augmented Backus-Naur Form (ABNF) notation of [[RFC5234]].
+
+The following core rules are included by reference, as defined in Appendix B.1 of [[RFC5234]]: 
+ALPHA (letters), CR (carriage return), CRLF (CR LF), CTL (controls), DIGIT (decimal 0-9), DQUOTE (double quote), 
+HEXDIG (hexadecimal 0-9/A-F/a-f), HTAB (horizontal tab), LF (line feed), OCTET (any 8-bit sequence of data), SP (space), 
+and VCHAR (any visible US-ASCII character).
+
+### 2.2. Requirements Notation
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", 
+"**SHOULD NOT**", "**RECOMMENDED**", "**NOT RECOMMENDED**", "**MAY**", and "**OPTIONAL**" 
+in this document are to be interpreted as described in BCP 14 [[RFC2119]] [[RFC8174]] when, 
+and only when, they appear in all capitals, as shown here.
+
+## 3. Terminology and Core Semantics
+### 3.1. VPM, VPM Client, VPM Repository, VPM Profile Folder, and VPM Package
+The term "VPM" refers to the VRChat Package Manager, which is the package manager system.
+The VPM consists of the VPM Client and the VPM Repository.
+
+The VPM Client is the software that manages the VPM Packages in the Unity project.
```

Comment:
I don't know best practices for RFC style documents, so my comment about using bulleted points does not apply here.